### PR TITLE
Remove user input for distro detection

### PR DESCRIPTION
**Pull request by alexandrasandulescu**
Add support in install/install.py for automatically distro detection

### CODE REVIEW

File: install/install.py [hunk 1]
```diff
@@ -77,28 +78,25 @@ def install_restricted_from_cfg(self, config_file):
...
+            status, result = commands.getstatusoutput(command)
+        distro_num = 0
+        if "kali" in result.lower():
+            distro_num = 1
+        elif "samurai" in result.lower():
+            distro_num = 2
+
+        if distro_num != 0:
+            self.run_command(cp.get(cp.sections()[int(distro_num)-1], "install"))
+        else:
```
Comment (tunnelshade) [inline install/install.py]: If we cannot detect the distro, please prompt the user to select one. There are cases when someones takes a distribution and then add few new sources and customizations and rename it or someone using debian but has all the kali repos. So, if we are unable to detect known distros prompt the user to select one

Comment (alexandrasandulescu) [inline install/install.py]: I have modified the commit and now it includes the changes you mentioned.

File: install/install.py [hunk 2]
```diff
@@ -77,12 +78,29 @@ def install_restricted_from_cfg(self, config_file):
             self.install_in_directory(os.path.expanduser(cp.get(section, "directory")), cp.get(section, "command"))
 
     def install(self):
-        # User asked to select distro and distro related stuff is installed
+        # User asked to select distro (in case it cannot be automatically detected) and distro related stuff is installed
         cp = ConfigParser.ConfigParser({"RootDir":self.RootDir, "Pid":self.pid})
         cp.read(self.distros_cfg)
 
+        #Try get the distro using lsb_release
+        status, result = commands.getstatusoutput("lsb_release -si")
```
Comment (tunnelshade) [inline install/install.py]: Commands module is removed in python3. It will cause portability issues. Use subprocess and we can merge this

**Comment by DePierre** [comment]
Maybe we could do that in python? I know, that's silly...
```python
import platform
platform.linux_distribution()
```
[Python 2.X](https://docs.python.org/2.7/library/platform.html#platform.linux_distribution) or [Python 3.X](https://docs.python.org/3.4/library/platform.html#platform.linux_distribution)

**Comment by tunnelshade** [comment]
Yup. Check [this](http://stackoverflow.com/questions/4553129/when-to-use-os-name-sys-platform-or-platform-system) for more info :P

**Comment by alexandrasandulescu** [comment]
I tried to use sys.platform() and platform.system() but I don't receive the distro, but the os type (as it says in description - Linux, Windows, etc).
os.uname() seems ok but I had a lot of problem with the distro it provides(for example on centos releases). The distribution must be parsed from the output also.

I suggest using subprocess to avoid compatibility issues and assure that the output is correct
What do you think about it?

**Comment by DePierre** [comment]
@alexandrasandulescu I think that if python provides such functions to get the distribution, and that such functions have been standard since python 2.x, we might want to use them instead of making our own soup.

I am opened to discussion.

**Comment by alexandrasandulescu** [comment]
hi @DePierre ,
I replaced command with subprocess as @tunnelshade  mentioned and the methods used now (check_call, check_output) are available since python 2.7 [1] . Does it break compatibility?
I will switch to 'uname ' if it does.
I am not against using it. I tested in kali and it prints the correct distro. I will have to test in samuraiwtf too.

Thanks

[1]https://docs.python.org/2/library/subprocess.html

**Comment by DePierre** [comment]
@alexandrasandulescu I have more faith in python's standard functions that gives you the distribution information than a home-made bash script.

Also, I don't like calling bash script from python source code. If we are writing python, let's write python.

Why would we prefer home-made bash script called from within python using ``subprocess`` manually than using python standard functions that do all the work already and have been tested since 10 years?

@tunnelshade What's your feeling about that?

**Comment by alexandrasandulescu** [comment]
@DePierre  @tunnelshade 
I used platform.linux_distribution()
How does it look now?
Thanks

**Comment by tunnelshade** [comment]
`platform.linux_distribution` reads from `/etc/lsb-release`. What about non debian based distros?

**Comment by DePierre** [comment]
@tunnelshade That would not be true. According to their [source code](https://hg.python.org/cpython/file/3.5/Lib/platform.py#l309):
```python
"""
        The function first looks for a distribution release file in
        /etc and then reverts to _dist_try_harder() in case no
        suitable files are found.
"""
```
Notice the "a distribution file" meaning not only ``/etc/lsb-release``. And with ``_dist_try_harder()`` implementing [other tricks](https://hg.python.org/cpython/file/3.5/Lib/platform.py#l203) to find the distribution.

If you meant that you want to check for MacOS or Windows as well, better support them first, and then add the checks imho.

**Comment by alexandrasandulescu** [comment]
According to [1] the supported distributions are 
'SuSE', 'debian', 'fedora', 'redhat', 'centos', 'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', gentoo', 'UnitedLinux', 'turbolinux'.

[1] https://hg.python.org/cpython/file/2.7/Lib/platform.py#l259

**Comment by tunnelshade** [comment]
In arch linux, it only returns 'arch' which might be painful in future.

@DePierre I read that statement from stackoverflow. I should have checked the source.

Not everything is covered under this linux_distribution method. Any ideas? @DePierre

**Comment by DePierre** [comment]
@tunnelshade my 2 cents: Python's standard functions will cover more cases than doing some home made tests.
As for the ``arch``, I don't see much of a big deal here. Even with only ``arch`` we can update the install script accordingly.

We can still rollback to ``platform.platform()`` later. We will have plenty of time before reaching the limits of ``platform.linux_distribution()``. One day we will have to add support for Windows and/or Mac.

I will test @alexandrasandulescu's patch.

**Comment by DePierre** [comment]
@alexandrasandulescu Would it be possible to tell the user that OWTF automatically detected X platform and is going to install automatically.

For instance something like ``[*] Kali has been automatically detected... Continuing in auto-mode`` or something. Otherwise, your PR is good to me :)

**Comment by alexandrasandulescu** [comment]
@DePierre done .

**Comment by DePierre** [comment]
@alexandrasandulescu Looks good to me and I am ready to merge it as-is. Nice feature you implement here :)

Any thought @tunnelshade?

**Comment by tunnelshade** [comment]
Ok :+1:

**Comment by DePierre** [comment]
@alexandrasandulescu Thank you for your contribution ;)

**Comment by alexandrasandulescu** [comment]
thank you for your help and guidance! @all

**Pull request merged**